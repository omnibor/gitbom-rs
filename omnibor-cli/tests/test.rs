use insta::Settings;
use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
use std::process::Command;

macro_rules! settings {
    ($block:expr) => {
        let mut settings = Settings::clone_current();
        settings.add_filter(r#"omnibor(?:\.exe)?"#, "omnibor");
        settings.bind(|| $block);
    };
}

#[test]
fn no_args() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")));
    });
}

#[test]
fn artifact_no_args() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).arg("artifact"));
    });
}

#[test]
fn manifest_no_args() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).arg("manifest"));
    });
}

#[test]
fn debug_no_args() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).arg("debug"));
    });
}
