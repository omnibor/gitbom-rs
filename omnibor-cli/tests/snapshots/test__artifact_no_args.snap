---
source: omnibor-cli/tests/test.rs
info:
  program: omnibor
  args:
    - artifact
---
success: false
exit_code: 2
----- stdout -----

----- stderr -----
Actions related to Artifact Identifiers

Usage: omnibor artifact [OPTIONS] [COMMAND]

Commands:
  id    For files, prints their Artifact ID. For directories, recursively prints IDs for all files under it
  find  Find file matching an Artifact ID
  help  Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help (see more with '--help')
  -V, --version  Print version

General Flags:
  -b, --buffer <BUFFER>  How many print messages to buffer at once [env: OMNIBOR_BUFFER=] [default: 100]
  -f, --format <FORMAT>  Output format [env: OMNIBOR_FORMAT=] [possible values: plain, json, short]
  -H, --hash <HASH>      Hash algorithm to use when parsing Artifact IDs [env: OMNIBOR_HASH=] [possible values: sha256]
  -d, --dir <DIR>        Directory to store manifests [env: OMNIBOR_DIR=]
